### PR TITLE
rocksdb: scan not fill block-cache

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
ref https://github.com/pingcap/tikv/issues/1244

@siddontang @BusyJay @disksing PTAL

### CODE REVIEW

File: src/storage/mvcc/reader.rs [hunk 1]
```diff
@@ -101,13 +101,13 @@ impl<'a> MvccReader<'a> {
         if self.scan_mode.is_some() {
             if self.write_cursor.is_none() {
                 self.write_cursor = Some(try!(self.snapshot
-                    .iter_cf(CF_WRITE, None, self.scan_mode.unwrap())));
+                    .iter_cf(CF_WRITE, None, false, self.scan_mode.unwrap())));
```
Comment (siddontang) [inline src/storage/mvcc/reader.rs]: why here is false but following is true?

File: src/storage/mvcc/reader.rs [hunk 2]
```diff
@@ -69,7 +69,7 @@ impl<'a> MvccReader<'a> {
     pub fn load_lock(&mut self, key: &Key) -> Result<Option<Lock>> {
         if self.scan_mode.is_some() && self.lock_cursor.is_none() {
             self.lock_cursor = Some(try!(self.snapshot
-                .iter_cf(CF_LOCK, None, self.scan_mode.unwrap())));
+                .iter_cf(CF_LOCK, None, false, self.scan_mode.unwrap())));
```
Comment (siddontang) [inline src/storage/mvcc/reader.rs]: seem that we have little locks key at same time, maybe we can use true here?

**Comment by siddontang** [comment]
LGTM

File: src/storage/mvcc/reader.rs [hunk 3]
```diff
@@ -49,7 +49,7 @@ impl<'a> MvccReader<'a> {
             return Ok(vec![]);
         }
         if self.scan_mode.is_some() && self.data_cursor.is_none() {
-            self.data_cursor = Some(try!(self.snapshot.iter(None, self.scan_mode.unwrap())));
+            self.data_cursor = Some(try!(self.snapshot.iter(None, false, self.scan_mode.unwrap())));
```
Comment (disksing) [inline src/storage/mvcc/reader.rs]: Should `fill_cache` be true for scan?

Comment (disksing) [inline src/storage/mvcc/reader.rs]: Ignore it, I figured it out.

**Comment by disksing** [comment]
LGTM

**Pull request merged**